src/middlewares/basicAuthMiddleware.js

```diff
@@ -0,0 +1,148 @@
+import client from 'Config/client'; // eslint-disable-line import/no-unresolved
+import crypto from 'crypto';
+import LesgoException from '../exceptions/LesgoException';
+import { errorHttpResponseAfterHandler } from './errorHttpResponseMiddleware';
+
+const FILE = 'Middlewares/basicAuthMiddleware';
+
+const blacklistMode = opts => {
+  if (opts && typeof opts.blacklistMode !== 'undefined') {
+    return !!opts.blacklistMode;
+  }
+
+  return true;
+};
+
+export const generateBasicAuthorizationHash = (key, secret) => {
+  return crypto
+    .createHash('sha1')
+    .update(`${key}:${secret}`)
+    .digest('hex');
+};
+
+const getSiteId = event => {
+  let siteId;
+
+  if (event.site && event.site.id) {
+    siteId = event.site.id;
+  } else if (
+    event.requestContext &&
+    event.requestContext.site &&
+    event.requestContext.site.id
+  ) {
+    siteId = event.requestContext.site.id;
+  } else if (event.platform) {
+    siteId = event.platform;
+  }
+
+  if (typeof siteId === 'undefined') {
+    throw new LesgoException(
+      'Site ID could not be found',
+      `${FILE}::SITE_ID_NOT_FOUND`,
+      403,
+      'Ensure that clientAuthMiddleware() is called before this Middleware'
+    );
+  }
+
+  return siteId;
+};
+
+const getClient = opts => {
+  if (opts && opts.client && Object.keys(opts.client).length > 0) {
```

Comment:
@jpaylaga This is more of a utility standpoint. But just an idea since you did it already as `Config/client.js`, can you please consider that as a source here as well, which is the intention right ?